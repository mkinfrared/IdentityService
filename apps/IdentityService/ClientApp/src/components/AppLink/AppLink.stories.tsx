import { Meta, Story } from "@storybook/react";

import { AppLink } from "./AppLink";
import css from "./AppLink.module.scss";
import { AppLinkProps } from "./AppLink.type";

export default {
  title: "COMPONENTS/AppLink",
  component: AppLink,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<AppLinkProps> = (args) => (
  <div className={css.Story}>
    <AppLink {...args} />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
