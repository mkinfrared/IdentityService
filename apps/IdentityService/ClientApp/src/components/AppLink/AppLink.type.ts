import { ReactNode } from "react";

import { MainRoutes } from "routes/MainRoute";

export type AppLinkProps = {
  children?: ReactNode;
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  to: MainRoutes;
};
