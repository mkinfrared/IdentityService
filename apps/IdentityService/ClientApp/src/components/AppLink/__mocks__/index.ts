export { default } from "./AppLink.mock";
