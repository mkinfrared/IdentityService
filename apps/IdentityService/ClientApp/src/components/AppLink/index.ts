export { default } from "./AppLink";

export * from "./AppLink.type";
