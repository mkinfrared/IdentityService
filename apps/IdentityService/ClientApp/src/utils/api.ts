import axios from "axios";

import { BASE_URL } from "utils/secrets";

const { isAxiosError } = axios;

const api = axios.create({
  baseURL: BASE_URL,
  withCredentials: true,
});

export { isAxiosError };

export default api;
