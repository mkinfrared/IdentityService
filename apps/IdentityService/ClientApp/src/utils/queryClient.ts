import { QueryClient } from "react-query";

const client = new QueryClient({
  defaultOptions: {
    queries: {
      suspense: true,
    },
  },
});

export default client;
