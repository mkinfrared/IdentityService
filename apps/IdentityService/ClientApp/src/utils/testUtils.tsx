import { ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "react-query";
import { MemoryRouter } from "react-router-dom";

type WithQueryProps = {
  children?: ReactNode;
};

const withRouter = ({ children }: WithQueryProps) => (
  <MemoryRouter>{children}</MemoryRouter>
);

const withQuery = ({ children }: WithQueryProps) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};

export { withRouter, withQuery };
