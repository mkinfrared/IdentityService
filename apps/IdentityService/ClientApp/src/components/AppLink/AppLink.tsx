import { memo } from "react";
import { Link, useLocation } from "react-router-dom";

import { AppLinkProps } from "./AppLink.type";

const AppLink = ({ children, className, to }: AppLinkProps) => {
  const { search, hash } = useLocation();

  const direction = {
    search,
    hash,
    pathname: to,
  };

  return (
    <Link className={className} data-testid="AppLink" to={direction}>
      {children}
    </Link>
  );
};

export { AppLink };

export default memo(AppLink);
