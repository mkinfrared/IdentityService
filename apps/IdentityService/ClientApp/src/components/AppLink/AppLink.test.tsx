import { render } from "@testing-library/react";

import { withRouter } from "utils/testUtils";

import { AppLink } from "./AppLink";

describe("<AppLink />", () => {
  const to = "/foo/bar";
  const Component = <AppLink to={to as any} />;

  it("should be defined", () => {
    expect(AppLink).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component, { wrapper: withRouter });

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component, { wrapper: withRouter });
    const element = getByTestId("AppLink");

    expect(element).toBeDefined();
  });
});
