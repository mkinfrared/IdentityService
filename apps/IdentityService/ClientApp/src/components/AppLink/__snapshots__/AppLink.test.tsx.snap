// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLink /> should match the snapshot 1`] = `
<div>
  <a
    data-testid="AppLink"
    href="/foo/bar"
  />
</div>
`;
